### PR TITLE
Chore: Script to sign extensions for Firefox

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- ~~Added/Updated related tests.~~

## Short description of the change(s)

The new script `sign-extension` automatically packages the extension and
uploads it for signing. The signed packaged is stored in
`web-ext-artifacts`.

In order for the script to work the following environment variables
need to exist and be valid:

* `WEB_EXT_API_KEY`
* `WEB_EXT_API_SECRET`

### CODE REVIEW

**Comment by molant** [comment]
@antross this will generate a new entry into the account each time it's called unless we reuse the id. I think for now we should be OK but maybe we want to have an extra step that changes the name to `webhint-development` or something like that so it's easier to identify the ones that are for development and the official published ones?

File: packages/extension-browser/package.json [hunk 1]
```diff
@@ -123,6 +123,7 @@
     "lint:dependencies": "node ../../scripts/lint-dependencies.js",
     "lint:md": "node ../../scripts/lint-markdown.js",
     "package-extension": "web-ext build -s ./dist/bundle -a ./dist --overwrite-dest && npm run lint-bundle-size",
+    "sign-extension": "web-ext sign -s ./dist/bundle",
```
Comment (antross) [inline packages/extension-browser/package.json]: Can we have this output in the `./dist` folder just like `package-extension` does (if so the `.gitignore` update can be removed)? Also does it need the `--overwrite-dest` flag as well?
```suggestion
    "sign-extension": "web-ext sign -s ./dist/bundle -a ./dist --overwrite-dest",
```

Comment (molant) [reply packages/extension-browser/package.json]: `--overwrite-dest` doesn't work in this case. Also this command doesn't require running `package-extension`

File: packages/extension-browser/package.json [hunk 2]
```diff
@@ -112,7 +112,7 @@
   "repository": "webhintio/hint",
   "scripts": {
     "build": "npm-run-all build:*",
-    "build-release": "npm run clean && npm run build:assets && tsc --inlineSourceMap false --removeComments true && webpack --env.release && npm run package-extension",
+    "build-release": "npm run clean && npm run build:assets && tsc --inlineSourceMap false --removeComments true && webpack --env.release",
```
Comment (antross) [inline packages/extension-browser/package.json]: If we remove `&& npm run package-extension` here, we need to add it to the end of the `test` script or we won't get bundle size validation.

**Pull request closed without merging**